### DISCUSSION TITLE
Scale up of operators in Rabbitmq as pods in a cluster

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#9118] [Questions]
We are creating Rabbitmq deployments using Operators and Cards as mentioned below,
1. RMQ cluster kubernetes operator (replica 1)
2. messaging topology operator (replica 1)
3. RMQ cluster (3 nodes using HA policy)
In RMQ cluster we are applying 3 replicas in statefulset to create get high availability using ha policy (1 master node and 2 mirror nodes).

Similar to RMQ cluster, can we scale up the rmq cluster kubernetes operator and messaging topology operator as well for high availability? 

This is for the condition where nodes containing these operators goes down or restarts and it takes a lot of time to restart the node.

**Comment by mkuratczyk** [comment]
There's little value in doing so, given the Operator is not required for the existing clusters to be available (it's only needed to modify them, delete them or deploy new cluster). So in most cases, the fact the Operator is unavailable for a few minutes is not a problem (paradoxically, could be more of a problem in some test env, where clusters are often deployed and then deleted for example).

But yes, you can do this, one of the instances will acquire a lease and will become the only active instance. The other instance(s) will serve as stand-by. If you decide to go down this path, you might be interested in the timeout settings related to the lease: https://github.com/rabbitmq/cluster-operator/pull/111

**Reply by Zerpet** [reply]
Adding to the answer: you can find all the environment variables available to tweak the cluster operator behaviour here: https://rabbitmq.com/kubernetes/operator/configure-operator-defaults.html
